### PR TITLE
add devcontainer.json config for .net 6 -Codespaces

### PR DESCRIPTION
**Pull request by seantleonard**
## Why this change?

Enhance the PR review experience in GitHub Codespaces. Currently, review productivity is hindered because the default Codespaces container only has .NET 8. (While there is work to add .net 8 to dab, this fix enables .net6 (current) compat in codespaces).

Pros:
- Everything that is needed is pre-installed in container: 
  - .NET 6 SDK -> allows you to "view definition" and other symbols with intellisense
  - C# Dev Kit VsCode extension -> view solution , run tests etc.
- No need to update this file when DAB is updated to build against .NET 8.

Cons:
- Slower to load CodeSpace because container needs to build (1-3 minutes), but once built, easy access for future PR iterations.

## What is this change?
- Docs used: 
  - https://docs.github.com/en/codespaces/setting-up-your-project-for-codespaces/adding-a-dev-container-configuration/setting-up-your-dotnet-project-for-codespaces
  -  https://github.com/devcontainers/features/tree/main/src/dotnet
- Adds folder `.devcontainer`
- Adds file `devcontainer.json` which
  - Uses .net 8 sdk image for future compatibility with DAB changes in #2140 .
  - Installs .net 6 SDK, Runtime, and ASPNETCORE
  - Installs C#devKit Vscode extension

## Actual experience
1.  When loading codespaces you'll see this container setup after selecting which codespace to create or reuse.
image
2. Auto install required extensions to make intellisense work:
image
3. Solution explorer works
image
4. Working intellisense!
image

### CODE REVIEW

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -0,0 +1,18 @@
+{
+    "name": ".NET 8 with .NET 6",
+    "image": "mcr.microsoft.com/dotnet/sdk:8.0",
+    "features": {
+		// https://github.com/devcontainers/features/tree/main/src/dotnet
```
Comment (seantleonard) [inline .devcontainer/devcontainer.json:5]: this comment links to the source of where I found which params to use for the snippet in lines 6-10

File: .devcontainer/devcontainer.json [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+        "ghcr.io/devcontainers/features/dotnet:2": {
+            "version": "latest",
+            "dotnetRuntimeVersions": "6.0",
+            "aspNetCoreRuntimeVersions": "6.0",
+			"additionalVersions": "6.0"
+        }
+    },
+	"customizations": {
```
Comment (seantleonard) [inline .devcontainer/devcontainer.json:13]: alignment looks off in GitHub code review UI, but in code editor is aligned: 
image

**Comment by seesharprun** [comment]
You can speed up the codespaces build by [pre-building](https://docs.github.com/en/codespaces/prebuilding-your-codespaces) the devcontainer on the repository. This is a trick I often use with various https://github.com/azure-samples repos.

**Comment by seesharprun** [comment]
If you need to test the .NET tool locally, you will need to add the .NET tools directory to the PATH which is not automatically done in the SDK repos:

```json
{
    ...,
    "remoteEnv": {
        "PATH": "${containerEnv:PATH}:/root/.dotnet/tools"
    }
}
```

**Review by Aniruddh25 — APPROVED** [review]
LGTM, thank you for adding innovative ways of improving productivity!

**Comment by severussundar** [comment]
Please add to the PR description the docs referenced for this change

**Comment by seantleonard** [comment]
azp run

**Pull request merged**